samples/otlptrace/example.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import google.auth
+from google.auth.transport.requests import AuthorizedSession
+
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+
+from opentelemetry import trace
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+
+credentials, project_id = google.auth.default()
+authed_session = AuthorizedSession(credentials)
+
+# Service name is required for most backends
+resource = Resource(attributes={
+    SERVICE_NAME: "your-service-name"
+})
+
+traceProvider = TracerProvider(resource=resource)
+processor = BatchSpanProcessor(OTLPSpanExporter(endpoint="<traces-endpoint>/v1/traces", headers=authed_session.headers))
```

Comment:
tested this out and it doesn't seem to contain the Authorization header:
```py
import google.auth
from google.auth.transport.requests import AuthorizedSession

credentials, project_id = google.auth.default()
print(credentials, project_id)
authed_session = AuthorizedSession(credentials)
print(authed_session.headers)
```

```sh
$ python test.py 
<google.oauth2.credentials.Credentials object at 0x7f2fbc5e2190> otel-starter-project
{'User-Agent': 'python-requests/2.31.0', 'Accept-Encoding': 'gzip, deflate, br', 'Accept': '*/*', 'Connection': 'keep-alive'}
```

Reply:
yeah this was wrong, updated to actually get the access token (I'm still getting a 401 but working on that separately)